flink-ml-lib/src/main/java/org/apache/flink/ml/common/linalg/SparseVector.java

```diff
@@ -0,0 +1,574 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.linalg;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/** A sparse vector represented by an indices array and a values array. */
+public class SparseVector extends Vector {
```

Comment:
If we decide to add `SparseVector`, it probably needs to have a `SparseVectorTypeInfoFactory` similar to the existing `DenseVectorTypeInfoFactory`, to achieve best possible performance for serialization and deserialization.

And we will also need to discuss the APIs of this class and make sure it is consistent with the DenseVector.

Can we start simple and only add minimum APIs needed by Knn in this PR?

Reply:
I have remove the sparse vector. this knn version only support dense vector calculate.